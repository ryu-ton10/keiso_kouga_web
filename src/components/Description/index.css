/* ========= メンバー紹介 =========== */
.description {
    position: relative;
    display: flex;
    flex-direction: column;
    justify-content: center;
    min-height: 800px;
    max-height: 1000px;
}

.description-title {
    height: 30%;
    text-align: center;
    font-family: 'Hiragino Mincho Pro';
    font-size: 36pt;
    color: rgba(105, 105, 105, 1);
}

.description-content {
    height: 70%;
    text-align: center;
    font-family: 'Hiragino Mincho Pro';
    font-size: 14pt;
    color: rgba(105, 105, 105, 1);
    animation-name: description-fade-in;
    animation-duration: 1.5s;
    animation-timing-function: ease-out;
    animation-fill-mode: forwards;
}
