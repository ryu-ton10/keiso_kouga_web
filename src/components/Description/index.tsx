import React from 'react';
import './index.css';

const Description: React.FC = () => {
    return (
        <div className="description">
            <div className="description-title">
                <p>珪素光画の世界へ、ようこそ</p>
            </div>
            <div className="description-content">
                <p>写真サークル 珪素光画は、日本の風景を写真を通してカタチにしています。<br/>
                荘厳な自然や身の回りの何気ない日常などを見ていると<br/>
                新たな発見をすることが出来るかも知れません。<br/>
                そんな体験を生むために、今日もカメラを構えます。</p>
            </div>
        </div>
    );
}

export default Description;
