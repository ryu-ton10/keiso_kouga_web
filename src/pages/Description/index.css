/* ========= サークルロゴ部分 =========== */
.logo {
  text-align: center;
  opacity: 1;
  margin: 100px 0 100px 0;
}

@media (min-width: 280px) {
  .logo {
    margin: 50px 0 50px 0;
  }
  .logo img {
    width: 150px;
    height: auto;
  }
}

@media (min-width: 960px) {
  .logo {
    margin: 80px 0 80px 0;
  }
  .logo img {
    width: 200px;
    height: auto;
  }
}

/* ========= サークル概要文 =========== */
.description {
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.description-title {
  height: 30%;
  text-align: center;
  font-family: 'Hiragino Mincho Pro';
}

.description-title > p {
  color: rgba(105, 105, 105, 1);
}

.description-content {
  height: 70%;
  text-align: center;
  font-family: 'Hiragino Mincho Pro';
  color: rgba(105, 105, 105, 1);
}

.description-content > p {
  color: rgba(105, 105, 105, 1);
}

/* ========= テキストのレスポンスデザイン =========== */
@media (min-width: 280px) {
  .description {
    min-height: 300px;
    max-height: 500px;
  }
  .description-title {
    font-size: 14pt;
  }
  .description-content {
    font-size: 8pt;
    margin-bottom: 20px;
  }
}

@media (min-width: 360px) {
  .description {
    min-height: 300px;
    max-height: 500px;
  }
  .description-title {
    font-size: 18pt;
  }
  .description-content {
    font-size: 9pt;
    margin-bottom: 30px;
  }
}

@media (min-width: 576px) {
  .description {
    min-height: 400px;
    max-height: 500px;
  }
  .description-title {
    font-size: 24pt;
  }
  .description-content {
    font-size: 12pt;
    margin-bottom: 50px;
  }
}

@media (min-width: 960px) {
  .description {
    min-height: 500px;
    max-height: 800px;
  }
  .description-title {
    font-size: 28pt;
  }
  .description-content {
    font-size: 14pt;
    margin-bottom: 70px;
  }
}

