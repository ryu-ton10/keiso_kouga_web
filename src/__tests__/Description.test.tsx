import renderer from 'react-test-renderer';
import Description from 'pages/Description';

/* ---------------- snapshot test -------------------- */
it('コンポーネントが描画されること', () => {
  const component = renderer.create(
    <Description />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})
