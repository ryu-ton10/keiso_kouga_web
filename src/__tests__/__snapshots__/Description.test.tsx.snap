// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`コンポーネントが描画されること 1`] = `
<div
  className="description"
>
  <div
    className="logo"
  >
    <picture>
      <source
        srcSet="keiso_kouga_logo.webp"
        type="image/webp"
      />
      <img
        alt="icon"
        src="keiso_kouga_logo.png"
      />
    </picture>
  </div>
  <div
    className="description-title"
  >
    <p>
      珪素光画の世界へ、ようこそ
    </p>
  </div>
  <div
    className="description-content"
  >
    <p>
      写真サークル 珪素光画は、日本の風景を主に撮影しています。
      <br />
      荘厳な自然や身の回りの何気ない日常などを見ていると
      <br />
      新たな発見をすることが出来るかも知れません。
      <br />
      そんな体験を生むために、今日もカメラを構えます。
    </p>
  </div>
</div>
`;
